frontend/src/components/Admin/AddUser/AddUser.tsx

```diff
@@ -125,6 +146,46 @@ export default function AddUser(): JSX.Element {
     });
   }
 
+  async function uploadUsersCsv(csvFile: File | undefined): Promise<void> {
+    if (csvFile) {
+      const csv = await csvFile.text();
+      const json = await csvtojson().fromString(csv);
+      const allNetIds = allMembers.map((m) => m.netid);
+      const members: IdolMember[] = [];
+      if (json[0].email && json[0].role) {
+        json.forEach((m) => {
+          const netId = getNetIDFromEmail(m.email);
+          const currMember = allMembers.find((mem) => mem.netid === netId);
+          if (currMember) {
+            const member = {};
+            Object.keys(EMPTY_MEMBER).forEach((field) => {
+              member[field] = m[field] ?? currMember[field];
+            });
+            members.push(member);
+          } else {
+            const member = { ...EMPTY_MEMBER };
+            Object.keys(EMPTY_MEMBER).forEach((field) => {
+              if (m[field]) member[field] = m[field];
+            });
+            members.push(member);
+          }
+        });
+        members.forEach((m) => {
```

Comment:
Hm, can we upload each member one by one by calling `MembersAPI.updateMember`/`setMember` in the `json.forEach` function? Instead of first populating a list, and then going through the list again to call the API for each one?